server/game/cards/attachments/02/lady.js

```diff
@@ -1,47 +1,60 @@
 const DrawCard = require('../../../drawcard.js');
 
 class Lady extends DrawCard {
+    constructor(owner, cardData) {
+        super(owner, cardData);
+        this.oldOwner = null;
+    }
+
     setupCardAbilities(ability) {
         this.whileAttached({
             effect: ability.effects.modifyStrength(2)
         });
         this.action({
             title: 'Pay 1 gold to attach Lady to another character',
-            method: 'reAttach'
+            method: 'reAttach',
+            limit: ability.limit.perPhase(1)
         });
     }
 
     reAttach(player) {
-        if(!this.parent || this.controller.gold < 1) {
+        this.oldOwner = this.parent;
+        if(!this.oldOwner || this.controller.gold < 1) {
             return false;
         }
 
-        player.removeAttachment(this, false);
-        player.promptForAttachment(this);
+        player.moveCard(this, 'play area');
 
-        player.gold -= 1;
-
-        this.triggered = true;
+        this.game.promptForSelect(this.controller, {
+            cardCondition: card => this.canAttach(player, card) && card.location === 'play area',
+            activePromptTitle: 'Select a different character for attachment',
+            waitingPromptTitle: 'Waiting for opponent to move attachment',
+            onSelect: (player, card) => this.moveAttachment(player, card)
+        });
 
         return true;
     }
 
     canAttach(player, card) {
-        if(!card.isFaction('stark')) {
+        if(!card.isFaction('stark') || card.getType() !== 'character' || card === this.oldOwner) {
             return false;
         }
 
         return super.canAttach(player, card);
     }
 
-    attach(player, card) {
-        if(this.triggered) {
-            if(card.name === 'Sansa Stark' && card.kneeled) {
-                player.standCard(card);
-            }
-
-            this.triggered = false;
+    moveAttachment(player, newOwner) {
+        var targetPlayer = this.game.getPlayerByName(newOwner.controller.name);
+        targetPlayer.attach(player, this, newOwner.uuid);
+        player.gold -= 1;
+        if(newOwner.name === 'Sansa Stark' && newOwner.kneeled) {
```

Comment:
You should set `this.oldOwner` to null here (and also in Nymeria). 

Scenario:
I use Lady to reattach to a different character (oldOwner set to original character).
Opponent uses something to remove Lady (goes back to your hand or you fish it out of discard later).
I try to play Lady again on the original character. I can't because of the added check in `canAttach` says I can't attach it to the old owner (which is still set).